auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -127,36 +127,38 @@ private suspend fun OidcConfiguration.executeRequest(
 ): Response {
     currentCoroutineContext().ensureActive()
     val rateLimitErrorCode = 429
+    var event: RateLimitExceededEvent
+    var retryCount = 0
+    var response: Response
+    var requestToExecute = request
 
-    var response = okHttpClient.newCall(request).await()
+    do {
+        response = okHttpClient.newCall(requestToExecute).await()
+        if (response.code != rateLimitErrorCode) return response
 
-    if (response.code == rateLimitErrorCode) {
-        val event = RateLimitExceededEvent(request, response)
+        event = RateLimitExceededEvent(requestToExecute, response, retryCount)
         eventCoordinator.sendEvent(event)
+        retryCount++
+
+        val timeToResetSeconds: Long
+        val requestId = response.header("X-Okta-Request-Id")
+        val responseTimeSeconds = response.headers.getDate("Date")?.time?.milliseconds?.inWholeSeconds
+        val retryTimeSeconds = response.header("X-Rate-Limit-Reset")?.toLongOrNull()
+        if (responseTimeSeconds == null || retryTimeSeconds == null) {
+            return response
```

Comment:
I don't think we want to send the event in this case. Can we postpone the event until after this?

Reply:
Good catch! Yes, I will do that